media-gfx/ahoviewer/ahoviewer-1.6.1.ebuild

```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+GNOME2_EAUTORECONF="yes"
+
+inherit gnome2 xdg-utils
+
+DESCRIPTION="A GTK2 image viewer, manga reader, and booru browser"
+HOMEPAGE="https://github.com/ahodesuka/ahoviewer"
+SRC_URI="https://github.com/ahodesuka/ahoviewer/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="gnutls +gstreamer libressl libsecret +rar ssl +zip"
+
+DEPEND="
+	dev-cpp/glibmm:2
+	dev-cpp/gtkmm:2.4
+	dev-libs/libconfig[cxx]
+	dev-libs/libxml2
+	dev-libs/libsigc++:2
+	net-misc/curl:=
```

Comment:
should not it be `[ssl=]` ?

Reply:
It needs curl, but doesnt require SSL to work. However that pycurl suggestion was a good one, I see the logic in that and am gonna try implementing it here as well later today. Thanks!

Reply:
correction.
that will quire ssl for curl if ssl flag is enabled. not sure if needed but may be.

```sh
app-misc/foo[bar?]

```

same as

```sh
bar? ( app-misc/foo[bar] ) !bar? ( app-misc/foo )
```

Reply:
but probably you need to play around curl_ssl_ things, not bare ssl. no prob, submit it and tear apart mgorny, he'll know this hell for sure =)

Reply:
Ahh, `?` instead of `=` makes sense. With `=` it won't allow de-selecting `ssl` at all. Think I'm getting there, I saw that support for LibreSSL has probably been dropped, so this got a bit easier. However, the ebuild is not cleaner. Here's what I currently came up with, 
```
IUSE="curl_ssl_gnutls +curl_ssl_openssl +gstreamer libsecret +rar +ssl +zip"
...
    ssl? (
        net-misc/curl[curl_ssl_gnutls(-)=,curl_ssl_openssl(-)=]
    )
...
src_configure() {
    local myconf=(
        $(use_enable gstreamer gst)
        $(use_enable libsecret)
        $(use_enable rar)
        $(use_enable zip)
    )

    if use ssl; then
        myconf+=(
            $(use_enable curl_ssl_gnutls gnutls)
            $(use_enable curl_ssl_openssl ssl)
        )
    else
        myconf+=(
            --disable-gnutls
            --disable-ssl
        )
    fi

    gnome2_src_configure "${myconf[@]}"
}
```

Im testing the logic right now, if it works fine, gonna submit.